master/internal/restore.go

```diff
@@ -298,3 +304,23 @@ func shimTrialSnapshotV0(snapshot []byte) ([]byte, error) {
 
 	return json.Marshal(trialSnapshotV0)
 }
+
+// Version 1 => 2 shims
+
+// shimExperimentSnapshotV1 shims a v1 snapshot to a v2 snapshot. From v1 to v2,
+// progress was rewritten so that individual trial progress was tracked and aggregated
+// instead of just maintaining the total units completed.
+func shimExperimentSnapshotV1(snapshot []byte) ([]byte, error) {
+	var experimentSnapshotV1 map[string]interface{}
+	if err := json.Unmarshal(snapshot, &experimentSnapshotV1); err != nil {
+		return nil, err
+	}
+
+	searcherState := experimentSnapshotV1["searcher_state"].(map[string]interface{})
+
+	delete(searcherState, "total_units_completed")
+	delete(searcherState, "units_completed_by_trial")
+	searcherState["trial_progress"] = map[model.RequestID]model.Length{}
```

Comment:
Will this cause experiments after master-restart to end at < 100% progress if some of those trials are finished at the time that the experiment is paused?

Reply:
yes, this would happen but only on the upgrade from 0.14.6 to the next version. I felt this was fine, but I supposed I could just use `units_completed_by_trial` to properly shim this.

Reply:
I'd like to avoid the "this experiment is stuck at 50% even though its completed" bug reports

Reply:
Ok i changed it.